src/routing/router.rs

```diff
@@ -10,16 +11,20 @@ pub struct Router<'router> {
 }
 
 impl<'router> Router<'router> {
-    pub(crate) fn new(cch: &'router CCH, graph: &OwnedGraph) -> Router<'router> {
+    pub(crate) fn new(cch: &'router CCH, converter: &NetworkConverter) -> Router<'router> {
         Router {
-            server: Server::new(customize(cch, graph))
+            server: Server::new(customize(cch, &Router::create_owned_graph(&converter)))
         }
     }
 
     pub(crate) fn customize(&mut self, cch: &'router CCH, graph: &OwnedGraph) {
         self.server = Server::new(customize(cch, graph));
     }
 
```

Comment:
Very nice, that you figured out the lifetimes. The signature looks terribly complicated though. Would it be possible to use a type alias for this?

Reply:
I'm not quite sure. We need the lifetime `'q`, so I tried this one here: 
```
type P<'q> = PathServerWrapper<'q, CustomizedBasic<'router, CCH>>;
pub(crate) fn query<'q>(
        &'q mut self,
        from: usize,
        to: usize,
    ) -> QueryResult<Self::P<'_>, Weight>
```
But the compiler complains `type must outlive the lifetime 'q`. Unfortunately, the explanation in the rust reference is not very detailed. So I don't really know what to try else.

Reply:
That's exactly what I meant. That makes sense, that we have to have the different lifetimes of the `PathServerWrapper` and the `CCH` and therefore have to keep it all in this signature. Thanks for trying!